### PR TITLE
[jsonrpc] Add boxset, disctitle & totaldiscs fields

### PR DESCRIPTION
**Pull request by the-black-eagle**
## Description
Adds the new fields in the music database to jsonrpc.
## Motivation and Context
This updates the jsonrpc interface to return `isboxset` and `totaldiscs` for an album and `disctitle` for songs. Allows filtering by `isboxset` true|false and sorting by `totaldiscs`. Also allows the boxset flag to be set or unset for an album via json but will not set it for albums with only one disc. If the discs don't already have a title then they will be created automatically.  Existing disc titles are retained.

## How Has This Been Tested?

Tested locally with python json calls to the API and printing the results to the screen.  Returned data checked against database values to ensure correct. Made several multi-disc albums into box sets and then reverted them all.  Attempted to make a single disc album a box set.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [X] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [X] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [X] All new and existing tests passed

### CODE REVIEW

**Comment by DaveTBlake** [comment]
Thanks for this addition to the JSON API following from the addition of a boxset/multi-disc facility to the music library by https://github.com/xbmc/xbmc/pull/16915

**Comment by the-black-eagle** [comment]
Rebased.

File: xbmc/interfaces/json-rpc/schema/version.txt [hunk 1]
```diff
@@ -1 +1 @@
-JSONRPC_VERSION 11.1.1
+JSONRPC_VERSION 11.2.1
```
Comment (DaveTBlake) [inline xbmc/interfaces/json-rpc/schema/version.txt]: That needs to be 11.2.0, patch number gets zeroed

Comment (the-black-eagle) [reply xbmc/interfaces/json-rpc/schema/version.txt]: Fixed.

**Review by DaveTBlake — APPROVED** [review]
Have tested out the JSON AudioLibrary calls for `GetAlbums`, `GetAlbumDetails`, `GetSongs`, `GetSongDetals`, `SetAlbumDetails`, `SetSongDetails`. 
The totaldiscs and isboxset can both be returned for albums,  used in filter rules and results sorted by totaldiscs. It's possible to set an album that has two or more discs to be a boxset. The disc title can be read and set for a song. 

All looks good, thanks for this follow-up work @the-black-eagle

**Pull request merged**